packages/typescript-private/device-server/src/browser-manager/installed-browser-finders.ts

```diff
@@ -0,0 +1,134 @@
+import { AndroidBrowserName, AndroidBrowserPackageNameMap, isAllowedAndroidBrowserName, Serial } from '@dogu-private/types';
+import { PrefixLogger } from '@dogu-tech/common';
+import { BrowserInstallation, BrowserInstallationFinder, BrowserInstallationFinderOptions } from '@dogu-tech/device-client-common';
+import { Adb } from '../internal/externals/index';
+import { logger } from '../logger/logger.instance';
+import { SeleniumManager } from './selenium-manager';
+
+export class SeleniumManagerBrowserInstallationFinder implements BrowserInstallationFinder {
+  constructor(private readonly seleniumManager: SeleniumManager) {}
+
+  match(options: BrowserInstallationFinderOptions): boolean {
+    return this.seleniumManager.matchForBrowser(options);
+  }
+
+  async find(options: BrowserInstallationFinderOptions): Promise<BrowserInstallation[]> {
+    return await this.seleniumManager.findBrowserInstallation(options);
+  }
+}
+
+export class AdbBrowserInstallationFinder implements BrowserInstallationFinder {
+  static readonly packageLinePattern = /^package:(?<package>.*)=(?<packageName>.*)$/;
+  static readonly packageVersionLinePattern = /^\s*versionName=(?<version>.*)\s*$/;
+
+  private readonly logger = new PrefixLogger(logger, '[AdbBrowserInstallationFinder]');
+
+  match(options: BrowserInstallationFinderOptions): boolean {
+    const { browserName, browserPlatform } = options;
+    if (browserPlatform !== 'android') {
+      return false;
+    }
+
+    return isAllowedAndroidBrowserName(browserName);
+  }
+
+  async find(options: BrowserInstallationFinderOptions): Promise<BrowserInstallation[]> {
+    const { browserName, browserPlatform, deviceSerial } = options;
+    if (!deviceSerial) {
+      throw new Error(`Device serial is required. Browser name: ${browserName}, browser platform: ${browserPlatform}`);
+    }
+
+    const browserInfos = await this.findAllBrowserInstallations(deviceSerial);
+    const filteredInfos = browserInfos.filter(({ browserName: name }) => name === browserName);
+    const browserInfosWithVersion = await this.findBrowserInstallationsWithVersion(deviceSerial, filteredInfos);
+    return browserInfosWithVersion;
+  }
+
+  private async findAllBrowserInstallations(deviceSerial: Serial): Promise<{ browserPackageName: string; browserName: AndroidBrowserName }[]> {
+    const browserNameMap = new Map(Object.entries(AndroidBrowserPackageNameMap).map(([key, value]) => [value, key]));
+    const { stdout } = await Adb.shell(deviceSerial, 'pm list packages -f');
```

Comment:
How about make function like Adb.getIntalledPackage() in adb.ts?

Reply:
please review https://github.com/dogu-team/dogu/pull/374/commits/cfbeb97273ce556397c44917cca02b5215bcd2a0